sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1255,6 +1255,173 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-38811 INSERT INTO on columns added with ALTER TABLE ADD COLUMNS: Positive tests") {
+    // There is a complex expression in the default value.
+    withTable("t") {
+      sql("create table t(i boolean) using parquet")
+      sql("alter table t add column s string default concat('abc', 'def')")
+      sql("insert into t values(true, default)")
+      checkAnswer(sql("select s from t where i = true"), Seq("abcdef").map(i => Row(i)))
```

Comment:
```suggestion
      checkAnswer(sql("select s from t where i = true"), Row("abcdef"))
```

Reply:
I converted these positive tests to use the `checkAnswer(spark.table("t"))` syntax you suggested previously. They read simpler now, thank you for suggesting that.